subprojects/core-api/src/main/java/org/gradle/api/file/FileContents.java

```diff
@@ -47,6 +49,23 @@ public interface FileContents {
      */
     Provider<String> getAsText();
 
+    /**
+     * Gets a provider of the entire file contents as a single String.
+     *
+     * See {@link #getAsText()} for details
+     *
+     * @param charset Charset to be used
+     * @return provider of the entire file contents as a single String.
+     * @since TODO
+     */
+    default Provider<String> getAsText(String charset) {
+        if (Charset.defaultCharset().toString().equals(charset)) {
+            return getAsText();
+        } else {
+            throw new IllegalArgumentException("");
```

Comment:
@oleg-nenashev Is this a WIP PR? If so, please mark it as draft.

Reply:
Sorry, marked as draft. Now the issue is fixed though